openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -73,9 +76,20 @@ def target(self):
             dot_client_id=self.request.GET.get('client_id'),
             require_https=self.request.is_secure(),
         )
-        return target_url if use_target_url else self.default_target
+
+        if use_target_url:
+            return target_url
+
+        if self.is_user_panel_admin:
+            site_config = get_current_site_configuration()
+            return site_config.site_values.get('PANEL_NOTIFICATIONS_BASE_URL', self.default_target)
+
+        return self.default_target
 
     def dispatch(self, request, *args, **kwargs):
+        if request.user.groups.filter(name=EDLY_PANEL_ADMIN_USERS_GROUP):
+            self.is_user_panel_admin = True
```

Comment:
why are querying in dispatch when we can call same code in the post method

Reply:
kindly debug this further.

Reply:
actually, the dispatch method is called before http action methods (get, post etc.), and the logout function (from django's built in auth) to logout the user is called in the dispatch method,  that's why we cannot get user data in get or post method. so the only solution is to go with dispatch like this.